From 9979ed8e1e7bfc36afda9e589c1bf8d8b66f3f3a Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Thu, 23 May 2019 14:38:10 +0800
Subject: [PATCH] ntp: fix avc denials for ntpd_t

Fix:
type=AVC msg=audit(1559180388.257:76): avc:  denied  { read } for
pid=175 comm="systemd-timesyn" name="state" dev="tmpfs" ino=12189
scontext=system_u:system_r:ntpd_t:s0-s15:c0.c1023
tcontext=system_u:object_r:systemd_networkd_var_run_t:s0 tclass=file
permissive=1
type=AVC msg=audit(1559180388.257:76): avc:  denied  { open } for
pid=175 comm="systemd-timesyn" path="/run/systemd/netif/state"
dev="tmpfs" ino=12189 scontext=system_u:system_r:ntpd_t:s0-s15:c0.c1023
tcontext=system_u:object_r:systemd_networkd_var_run_t:s0 tclass=file
permissive=1
type=AVC msg=audit(1559180388.257:77): avc:  denied  { getattr } for
pid=175 comm="systemd-timesyn" path="/run/systemd/netif/state"
dev="tmpfs" ino=12189 scontext=system_u:system_r:ntpd_t:s0-s15:c0.c1023
tcontext=system_u:object_r:systemd_networkd_var_run_t:s0 tclass=file
permissive=1

Upstream-Status: Pending

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/services/ntp.te | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/policy/modules/services/ntp.te b/policy/modules/services/ntp.te
index bb29559..f17ed06 100644
--- a/policy/modules/services/ntp.te
+++ b/policy/modules/services/ntp.te
@@ -88,6 +88,9 @@ manage_dirs_pattern(ntpd_t, ntpd_tmpfs_t, ntpd_tmpfs_t)
 manage_files_pattern(ntpd_t, ntpd_tmpfs_t, ntpd_tmpfs_t)
 fs_tmpfs_filetrans(ntpd_t, ntpd_tmpfs_t, { dir file })
 
+allow ntpd_t system_dbusd_var_run_t:dir read;
+allow ntpd_t system_dbusd_var_run_t:sock_file read;
+
 can_exec(ntpd_t, ntpd_exec_t)
 
 kernel_read_kernel_sysctls(ntpd_t)
@@ -120,6 +123,7 @@ mls_file_read_all_levels(ntpd_t)
 
 files_manage_etc_symlinks(ntpd_t)
 files_read_etc_runtime_files(ntpd_t)
+files_map_etc_files(ntpd_t)
 files_read_usr_files(ntpd_t)
 files_list_var_lib(ntpd_t)
 
@@ -158,6 +162,7 @@ ifdef(`init_systemd',`
 
 	# for /run/systemd/netif/links
 	systemd_list_networkd_runtime(ntpd_t)
+	systemd_read_networkd_runtime(ntpd_t)
 
 	optional_policy(`
 		chronyd_enabledisable(ntpd_t)
-- 
2.7.4

